pandas/tests/tseries/offsets/test_ticks.py

```diff
@@ -35,6 +36,39 @@ def test_delta_to_tick():
     assert (tick == offsets.Day(3))
 
 
+@given(cls=st.sampled_from(tick_classes),
+       n=st.integers(-999, 999),
+       m=st.integers(-999, 999))
+def test_tick_add_sub(cls, n, m):
+    # For all Tick subclasses and all integers n, m, we should have
+    # tick(n) + tick(m) == tick(n+m)
+    # tick(n) - tick(m) == tick(n-m)
+    left = cls(n)
+    right = cls(m)
+    expected = cls(n + m)
+
+    assert left + right == expected
+    assert left.apply(right) == expected
+
+    expected = cls(n - m)
+    assert left - right == expected
+
+
+@given(cls=st.sampled_from(tick_classes),
```

Comment:
do these generate individual tests like pytest params?

Reply:
No, it's only one test at runtime - `@given(...)` creates a wrapper which handles the details of repeatedly calling the inner test.  It also shrinks each error it sees to the minimal example, and will report multiple errors if it saw more than one (distinguished by exception type and line of code).